File: client/app/types/review.ts
export interface Review {
    id: number
    user_id: number
    product_id: number
    purchase_id: number
    rating: number
    title_ru: string | null
    title_en: string | null
    comment_ru: string | null
    comment_en: string | null
    is_approved: boolean
    is_visible: boolean
    created_at: string
    updated_at: string
    status?: string // computed status: 'pending', 'approved', 'rejected'

    // Joined fields
    user?: {
        id: number
        username: string
        full_name: string | null
        email: string | null
    }
    product?: {
        id: number
        title_ru: string
        title_en: string
    }
    purchase?: {
        id: number
        product_id: number
    }
}

export interface CreateReviewDto {
    rating: number
    title_ru?: string
    title_en?: string
    comment_ru?: string
    comment_en?: string
}

export interface UpdateReviewDto {
    rating?: number
    title_ru?: string
    title_en?: string
    comment_ru?: string
    comment_en?: string
}

File: server/models/review.go
package models

import (
	"time"
)

type Review struct {
	Id         int       `json:"id"`
	UserId     int       `json:"user_id"`
	ProductId  int       `json:"product_id"`
	PurchaseId int       `json:"purchase_id"`
	Rating     int       `json:"rating"` // 1-5
	TitleRu    string    `json:"title_ru,omitempty"`
	TitleEn    string    `json:"title_en,omitempty"`
	CommentRu  string    `json:"comment_ru,omitempty"`
	CommentEn  string    `json:"comment_en,omitempty"`
	IsApproved bool      `json:"is_approved"` // модерация админом
	IsVisible  bool      `json:"is_visible"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`

	// Joined fields
	User     *User     `json:"user,omitempty"`
	Product  *Product  `json:"product,omitempty"`
	Purchase *Purchase `json:"purchase,omitempty"`
}
